charactersheet/charactersheet/services/character/armor_class_service.js

```diff
@@ -0,0 +1,171 @@
+        }
+
+        if (self.hasArmor()) {
+            calculatedAC += armorMagicalModifier;
+        }
+
+        // Set the value and let everyone know.
+        self.armorClass(calculatedAC);
+        Notifications.armorClass.changed.dispatch();
+    };
+
+    /* Public Methods */
+
+    self.baseArmorClass = function() {
+        var ac = 10;
+
+        var characterId = CharacterManager.activeCharacter().key();
+        var predicates = [
+            new KeyValuePredicate('characterId', characterId),
+            new KeyValuePredicate('armorEquipped', 'equipped'),
+            new NotPredicate(new KeyValuePredicate('armorType', 'Shield'))
+        ];
+
+        var armor = PersistenceService.findByPredicates(Armor, predicates);
+
+        if (armor.length > 0) {
+            return parseInt(armor[0].armorClass().trim().split(/[^0-9]/)[0]);
+        }
+        return ac;
+    };
+
+    self.equippedArmorMagicalModifier = function() {
+        var magicalModifier = 0;
+
+        var characterId = CharacterManager.activeCharacter().key();
+        var predicates = [
+            new KeyValuePredicate('characterId', characterId),
+            new KeyValuePredicate('armorEquipped', 'equipped'),
+            new NotPredicate(new KeyValuePredicate('armorType', 'Shield'))
+        ];
+
+        var armor = PersistenceService.findByPredicates(Armor, predicates);
+
+        if (armor.length > 0) {
+            magicalModifier = parseInt(armor[0].armorMagicalModifier());
+        }
+        return magicalModifier;
+    };
+
+    self.equippedShieldMagicalModifier = function() {
+        var magicalModifier = 0;
+
+        var characterId = CharacterManager.activeCharacter().key();
+        var predicates = [
+            new KeyValuePredicate('characterId', characterId),
+            new KeyValuePredicate('armorEquipped', 'equipped'),
+            new KeyValuePredicate('armorType', 'Shield')
+        ];
+
+        var shields = PersistenceService.findByPredicates(Armor, predicates);
+
+        if (shields.length > 0) {
+            magicalModifier = parseInt(shields[0].armorMagicalModifier());
+        }
+
+        return magicalModifier;
+    };
+
+    self.atLeastOneArmorEquipped = function() {
+        var characterId = CharacterManager.activeCharacter().key();
+        var predicates = [
+            new KeyValuePredicate('characterId', characterId),
+            new KeyValuePredicate('armorEquipped', 'equipped')
+        ];
+        var equippedArmors = PersistenceService.findByPredicates(Armor, predicates);
+        return equippedArmors.length > 0;
+    };
+
+    self.dexBonus = function() {
+        var characterId = CharacterManager.activeCharacter().key();
+        var rawDexBonus = 0;
+        try {
+            var scores = PersistenceService.findFirstBy(AbilityScores, 'characterId', characterId);
+            rawDexBonus = scores.modifierFor('Dex');
+        } catch(err) { /*Ignore*/ }
+
+        if (score) {
```

Comment:
Where does this variable come from? It's used here, but declared a couple lines down.